test/integration_test/basic_test.go

```diff
@@ -305,34 +313,107 @@ func BasicCsiRegression(tc *types.TestCase) func(*testing.T) {
+			t.Skip()
+		}
+
+		var err error
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
+
+		// Create and validate StorageCluster
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Validate Stork is enabled by default
+		require.True(t, cluster.Spec.Stork.Enabled, "failed to validate Stork is enabled by default, it should be enabled, but it is set to %v", cluster.Spec.Stork.Enabled)
 
-		// Sleep for 20 seconds to let operator start the update process
-		logrus.Debug("Sleeping for 20 seconds...")
-		time.Sleep(20 * time.Second)
+		// Validate Webhook controller arg doesn't exist by default
+		require.Empty(t, cluster.Spec.Stork.Args["webhook-controller"], "failed to validate webhook-controller, it shouldn't exist by default, but it is set to %s", cluster.Spec.Stork.Args["webhook-controller"])
 
-		logrus.Infof("Validate StorageCluster %s", cluster.Name)
-		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateUpdateTimeout, ci_utils.DefaultValidateUpdateRetryInterval, true, "")
+		// Validate HostNetwork is <nil> by default
+		require.Nil(t, cluster.Spec.Stork.HostNetwork, "failed to validate HostNetwork, it should be nil by default, but it isn't", cluster.Spec.Stork.HostNetwork)
+
+		logrus.Info("Delete stork pods and validate they get re-deployed")
+		err = appsops.Instance().DeleteDeploymentPods("stork", cluster.Namespace, 120*time.Second)
 		require.NoError(t, err)
 
-		logrus.Info("Enable CSI and validate StorageCluster")
-		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+		logrus.Info("Delete stork-scheduler pods and validate they get re-deployed")
+		err = appsops.Instance().DeleteDeploymentPods("stork-scheduler", cluster.Namespace, 120*time.Second)
 		require.NoError(t, err)
-		cluster.Spec.FeatureGates = map[string]string{"CSI": "true"}
-		cluster, err = ci_utils.UpdateStorageCluster(cluster)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
 		require.NoError(t, err)
 
-		// Sleep for 20 seconds to let operator start the update process
-		logrus.Debug("Sleeping for 20 seconds...")
-		time.Sleep(20 * time.Second)
-
-		logrus.Infof("Validate StorageCluster %s", cluster.Name)
-		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateUpdateTimeout, ci_utils.DefaultValidateUpdateRetryInterval, true, "")
-		require.NoError(t, err)
+		logrus.Info("Enable Stork webhook-controller and validate StorageCluster")
+		// At this point this map should be <nil>
+		if cluster.Spec.Stork.Args == nil {
+			cluster.Spec.Stork.Args = make(map[string]string)
+		}
+		cluster.Spec.Stork.Args["webhook-controller"] = "true"
+		cluster = ci_utils.UpdateAndValidateStork(cluster, ci_utils.PxSpecImages, ci_utils.K8sVersion, t)
```

Comment:
`PxSpecImages` and `K8sVersion` are initialized only once for a test run so if we don't change them during the test I think we could remove those 2 parameters and use them directly in the function instead. if future we may have modified images and k8s versions to validate during one test we should be good.

Reply:
We use different `PxSpecImages` for `BasicUpgrade` test already and for the K8sVersion, I already tried and got `import cycle not allowed` thats why I decided to pass them there. Also we are going to implement k8s upgrade so we would need to move it back anyways or else where.